users/views.py

```diff
@@ -0,0 +1,46 @@
+from django.shortcuts import render
+from django.views import View
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from django.http import Http404
+from users.models import User
+from users.serializers import UserSerializer
+# Create your views here.
+class UserApi(APIView):
```

Comment:
and instead of in each endpoint you check the request user is the same user you want to update you can just create `permissions.py` file and the following:
```python
class UserPermission(permissions.BasePermission):
    def has_object_permission(self, request, view, obj):
        # Read permissions are allowed to any request,
        # so we'll always allow GET, HEAD or OPTIONS requests.
        if request.method in permissions.SAFE_METHODS:
            return True

        # Instance must have an attribute named `owner`.
        return obj == request.user
```